### PR TITLE
integrate modal into appointmentList

### LINKED ISSUE #19: Create Dashboard View
- [x] Create basic dashboard view just as a template
- [x] Later incorporate appointment card onto it, modal, etc

### LINKED ISSUE #42: Connect Card and Modal Component
- [x] Connect EDIT Button on Card Component to Modal Component
- [x] Update App.tsx to route `/dashboard` to Card instead of Modal
- [x] Test Modal to see if values added are being retrieved on backend and updating the Card Component accordingly

### LINKED ISSUE #44: Turn on JWT auth for appointment routes
- [x] Turn on JWT auth for appoint routes
- [x] Set all appointment API calls to have `{withCredentials: true}` accordingly

### PR DESCRIPTION
**Pull request by pmar0**
- added modalAddView, a view that allows routing to have modal over dash (easiest way to trigger modal, also allows for better link sending)
- added placeholder "loading..." for appointmentList when loading and also check to see if there's no appointments, which displays a different message
- tweaked card to ensure it works with appointmentList
- set up modal error states and error onChange clearing
- implemented proper error catching from server + display to show validations (this presented a problem, will list below)
- eliminated much of the modal boolean trigger functionality, as we are using routing instead
- installed cookie-parser on server end and imported into server.ts in order to get JWT auth working
- changed export for JWT auth so it would work properly
- changed app controller bad responses, were code 500 and now are 400
- added auth requirement to all appointment paths and made sure that all current axios requests have "{withCredentials: true}" when needed. This is fully functional from my testing of display (get all) and create (post)
- added an appointmentPost interface to fix TS error that was complaining that there is no _id, as that was added in the previous PR

Major issue is  validations on the modal result in it going all skinny n weird due to block styling of validation errors making the modal too tall, maybe? Not sure, poked at it for all too long but Ima have to leave that to y'all. I can swing back to integrate edit and delete pretty quickly, but I just want that modal working first. @kishanpatel97 you might have an easier time figuring it out since you created the modal. **I currently have the alerts in as spans so things dont break immediately. Just change the validation spans back to p tags and you'll see what I mean.**

Resolves #44 not going to consider it solving the connection of card to modal yet, although that's an easy step.

Edit, added more because of everything that went on, see below:
commit d6f565f

- fixed modal validation issue 
- added edit and delete functionality
- set up edit functionality to default to add in case of error

Resolves #42 #19

### CODE REVIEW

File: client/src/components/appointmentList.tsx [hunk 1]
```diff
@@ -12,8 +11,9 @@ const AppointmentList: React.FC = (props) => {
   
   const fetchAppointments = async() => {
     try {
-      const response =  await axios.get<IAppointment[]>('http://localhost:8000/api/appointments')
+      const response =  await axios.get<IAppointment[]>('http://localhost:8000/api/appointments',{withCredentials:true})
       setAppointments(prev => [...prev, ...response.data])
+      console.log(response.data)
```
Comment (b-chang) [inline client/src/components/appointmentList.tsx]: should probably remove this

Comment (b-chang) [reply client/src/components/appointmentList.tsx]: or did i leave this in?

File: client/src/components/modal.tsx [hunk 2]
```diff
@@ -1,16 +1,23 @@
 import axios from 'axios';
 import React, { ChangeEvent, FormEvent, useRef, useState } from 'react';
-import { IAppointment } from '../../../shared/interfaces/appointment.interface';
+import { useNavigate } from 'react-router';
+import { IAppointmentPost } from '../../../shared/interfaces/appointment.interface';
 
 const Modal: React.FC = () => {
-    const [showModal, setShowModal] = useState(false);
-    const [errors, setErrors] = useState({});
+	const navigate = useNavigate();
```
Comment (b-chang) [inline client/src/components/modal.tsx]: tabbing

File: client/src/components/modal.tsx [hunk 3]
```diff
@@ -40,122 +51,126 @@ const Modal: React.FC = () => {
                     location: '',
                     notes: '',
                 });
-                setErrors('');
+                setErrors({
```
Comment (b-chang) [inline client/src/components/modal.tsx]: is this the actually tabbing? or is this just github being weird

**Review by b-chang — CHANGES_REQUESTED** [review]

File: client/src/components/appointmentList.tsx (hunk 1 above)
Comment (pmar0) [reply client/src/components/appointmentList.tsx]: I added when testing, forgot to remove last night..removed.

File: client/src/components/modal.tsx (hunk 2 above)
Comment (pmar0) [reply client/src/components/modal.tsx]: navigate actually somehow had a tab less..VSCode being weird

File: client/src/components/modal.tsx (hunk 3 above)
Comment (pmar0) [reply client/src/components/modal.tsx]: think I fixed...more VSCode tabbing fun!

**Review by kishanpatel97 — APPROVED** [review]
Looks hawt

**Review by kishanpatel97 — APPROVED** [review]
looks good. reformatted

**Pull request merged**